### PR TITLE
TPM feature page update

### PR DESCRIPTION
**Pull request by mz-pdm**
Changes proposed in this pull request:

- TPM feature page update according to the recent progress.

I confirm that this pull request was submitted according to the [contribution guidelines](https://github.com/oVirt/ovirt-site/blob/master/CONTRIBUTING.md): @mz-pdm 

This pull request needs review by: @ahadas @nyoxi @smelamud

### CODE REVIEW

**Review by nyoxi — APPROVED** [review]

File: source/develop/release-management/features/virt/tpm-device.md [hunk 1]
```diff
@@ -91,12 +91,26 @@ The options to deliver updated TPM data are similar to obtaining TPM data on VM
...
+
+After considering all the options discussed in the following subsections, the following solution was selected:
+
+- TPM data is stored in the Engine database, in a separate table.
+- A new parameter providing TPM data is added to VM.create call.
+- A new API call VM.getExternalData to retrieve TPM data is introduced.
+- Engine calls VM.getExternalData to update the data in the database after a VM is stopped but before it is destroyed.
+- If a VM is powered off rather than shut down from Engine or from within the guest, the latest available stable data is retrieved.  It may not correspond to the very final data, but this is always a risk with unclean shutdown.
+- TPM data for VM snapshots is stored in `snapshots` database table.
+- On VM export / import, the data is stored to / retrieved from OVF of the exported / imported VM.
```
Comment (ahadas) [inline source/develop/release-management/features/virt/tpm-device.md:109]: I think we still have an open question here - whether to add it to the OVF or treat it as a separate resource in the OVA (as we do with disks but the OVF specification also suggests that for icons, screenshots and such data)

File: source/develop/release-management/features/virt/tpm-device.md [hunk 2]
```diff
@@ -91,12 +91,26 @@ The options to deliver updated TPM data are similar to obtaining TPM data on VM
...
-Different options are discussed in the following subsections.  We can probably assume that TPM data updates are not going to be very frequent.  We mustn't forget about reboots, snapshots, imports and exports.
+We can assume that TPM data updates are not going to be very frequent.  We mustn't forget about reboots, snapshots, imports and exports.
+
+After considering all the options discussed in the following subsections, the following solution was selected:
+
+- TPM data is stored in the Engine database, in a separate table.
+- A new parameter providing TPM data is added to VM.create call.
+- A new API call VM.getExternalData to retrieve TPM data is introduced.
+- Engine calls VM.getExternalData to update the data in the database after a VM is stopped but before it is destroyed.
+- If a VM is powered off rather than shut down from Engine or from within the guest, the latest available stable data is retrieved.  It may not correspond to the very final data, but this is always a risk with unclean shutdown.
```
Comment (ahadas) [inline source/develop/release-management/features/virt/tpm-device.md:107]: right, and we mentioned the possibility to change power-off so it won't un-define the domain to ensure we'll get the latest data - the question is whether or not we can do it it a backward compatible way (I think we can)

File: source/develop/release-management/features/virt/tpm-device.md [hunk 3]
```diff
@@ -91,12 +91,26 @@ The options to deliver updated TPM data are similar to obtaining TPM data on VM
...
+- TPM data for VM snapshots is stored in `snapshots` database table.
+- On VM export / import, the data is stored to / retrieved from OVF of the exported / imported VM.
+
+The same approach can be used for storing secure boot NVRAM data.
+
+In a future cluster version, TPM data transferred to Engine is going to be compressed.
```
Comment (ahadas) [inline source/develop/release-management/features/virt/tpm-device.md:113]: so how about adding a section of future work with that and changing the power-off operation

**Review by ahadas — CHANGES_REQUESTED** [review]

File: source/develop/release-management/features/virt/tpm-device.md (hunk 1 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/tpm-device.md]: The two options added and briefly explained. I also added a note that TPM data is not imported from VMs originated from external systems.

File: source/develop/release-management/features/virt/tpm-device.md (hunk 2 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/tpm-device.md:107]: Explained in new Future work section.

File: source/develop/release-management/features/virt/tpm-device.md (hunk 3 above)
Comment (mz-pdm) [reply source/develop/release-management/features/virt/tpm-device.md]: Done.

**Review by ahadas — APPROVED** [review]

**Review by nyoxi — APPROVED** [review]

**Pull request merged**